website/src/components/inputTable/FunctionsTable.jsx

```diff
@@ -1,35 +1,45 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
 import HoverVisible from "../hoverVisible/HoverVisible";
-import { Checkbox, TextField } from "@mui/material";
+import { Checkbox} from "@mui/material";
+import { isNotNull, isSnakeCaseCapital } from "../../utils/validatorUtils";
+import InputString from "../inputs/InputString";
 
 const FunctionsTable = (props) => {
+    const [functions, setFunctions] = useState([]);
     const [visible, setVisible] = useState(false);
     const items = [];
     const isDataAccess = props.fieldKey.includes('core_lib.data_accesses.');
-    items.push(
+    useEffect(() => {
+        if(functions.length !== props.value.length){
+            setFunctions(props.value)
+        }
         
-        props.value.map((func, index) => {
-            
+    }, [props.value.length])
+    items.push(
+        functions.map((func, index) => {
```

Comment:
`functions.map` will return an array. and items are already an array.

can we just assign `items  = functions.map((fun....` ?